plugins/mtd/fu-mtd-device.c

```diff
@@ -293,19 +313,53 @@ fu_mtd_device_probe(FuDevice *device, GError **error)
 	if (attr_name != NULL)
 		fu_device_set_name(FU_DEVICE(self), attr_name);
 
-	/* set vendor ID as the BIOS vendor */
-	vendor = fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_MANUFACTURER);
-	fu_device_build_vendor_id(device, "DMI", vendor);
+	/* MTD devices backed by PCI should use that for identification */
+	parent_device = fu_device_get_backend_parent_with_subsystem(FU_DEVICE(self), "pci", NULL);
+	if (parent_device != NULL) {
+		self->is_pci_device = TRUE;
+
+		fu_device_incorporate(
+		    FU_DEVICE(self),
+		    parent_device,
+		    FU_DEVICE_INCORPORATE_FLAG_VENDOR | FU_DEVICE_INCORPORATE_FLAG_VENDOR_IDS |
+			FU_DEVICE_INCORPORATE_FLAG_VID | FU_DEVICE_INCORPORATE_FLAG_PID |
+			FU_DEVICE_INCORPORATE_FLAG_PHYSICAL_ID);
+
+		/* B&R PCI vendor id */
+		if (fu_device_get_vid(parent_device) == 0x1677) {
+			fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_USABLE_DURING_UPDATE);
+			fu_device_set_version_format(FU_DEVICE(self), FWUPD_VERSION_FORMAT_PAIR);
+			fu_device_set_version_raw(
+			    FU_DEVICE(self),
+			    fu_pci_device_get_revision(FU_PCI_DEVICE(parent_device)));
+		}
```

Comment:
this needs to be in the .quirk file really -- e.g.

    [PCI\VEN_1677]
    VersionFormat = pair

Could you also copy over the PCI `version_raw` unconditionally if the `version` is NULL for a PCI device. That seems useful, and a bit less magic. I'm not sure what `fu_pci_device_get_revision()` has that `fu_device_get_version_raw()` doesn't.

Reply:
I think merge this, and then I think it's worth trying to kill `fu_pci_device_get_revision()` in a new PR.